pallets/ceres-launchpad/src/lib.rs

```diff
@@ -97,7 +97,6 @@ pub mod pallet {
 
     const PALLET_ID: PalletId = PalletId(*b"crslaunc");
 
-    // TODO: #395 use AssetInfoProvider instead of assets pallet
     #[pallet::config]
     pub trait Config:
         frame_system::Config
```

Comment:
You should also remove `assets::Config` supertrait from `Config` trait in all pallets

Reply:
@vovac12 
I think we can't really do that for the following reasons: 
1. AssetInfoProvider doesn't cover all functionalities of asset pallet as we might need to add new trait Like AssetManager for methods like `transfer` and `mint` (because those methods are not related to asset info)

2. Also removing assets supertrait means we might need to move so many types from `assets::Config` to `common::Config` which still something to discuss. @wer1st

Reply:
`assets::Config` should one of the last dependencies that we can remove. It is necessary to remove dependencies to other pallets and after that we can remove `assets`

Reply:
But you need to try to remove `assets::Config` from pallet configs as much as you can. If pallet doesn't call `transfer`, `mint` etc. - you have to delete `assets::Config`.
Please go throw all pallets and check if you can remove `assets::Config`.
And don't forget to remove the `assets` dependency from the `Cargo.toml` if you remove `assets::Config` from the pallet config.

Reply:
We have added `AssetInfoProvider` for 12 pallets. I will provide each pallet along with the use of assets pallet outside the boundary of `AssetInfoProvider`:

- `ceres-launchpad`: https://github.com/sora-xor/sora2-network/blob/395-tech-debt-use-assetinfoprovider-instead-of-pallet/pallets/ceres-launchpad/src/lib.rs#L475
   here `transfer_from` and `burn` are used 
   
- `ceres-token-locker`: https://github.com/sora-xor/sora2-network/blob/395-tech-debt-use-assetinfoprovider-instead-of-pallet/pallets/ceres-token-locker/src/lib.rs#L222
   here `transfer_from` is used
   
 - `demeter-farming-platform`: https://github.com/sora-xor/sora2-network/blob/395-tech-debt-use-assetinfoprovider-instead-of-pallet/pallets/demeter-farming-platform/src/lib.rs#L1087
   here `transfer_from` and `mint` are used     
   
 - `eth-bridge`: https://github.com/sora-xor/sora2-network/blob/395-tech-debt-use-assetinfoprovider-instead-of-pallet/pallets/eth-bridge/src/lib.rs#L1525
   here `register_from`, `mint_to` and `gen_asset_id_from_any` are used        
   
  -  `hermes-governance-platform`: https://github.com/sora-xor/sora2-network/blob/395-tech-debt-use-assetinfoprovider-instead-of-pallet/pallets/hermes-governance-platform/src/lib.rs#L339
     here `transfer_from` is used
     
  - `liquidity-proxy`: https://github.com/sora-xor/sora2-network/blob/395-tech-debt-use-assetinfoprovider-instead-of-pallet/pallets/liquidity-proxy/src/lib.rs#L2004
    here `transfer_from`, `mint_to`, `burn_from` are used
   
  - `multicollateral-bonding-curve-pool`:  https://github.com/sora-xor/sora2-network/blob/395-tech-debt-use-assetinfoprovider-instead-of-pallet/pallets/multicollateral-bonding-curve-pool/src/lib.rs#L686
    here `transfer_from`, `mint_to`, `burn_from` are used
    
  - `pool-xyk`: here only used AssetId and AssetIdOf types from assets directly
  
  - `pswap-distribution`:  https://github.com/sora-xor/sora2-network/blob/395-tech-debt-use-assetinfoprovider-instead-of-pallet/pallets/pswap-distribution/src/lib.rs#L147
      here `transfer_from`, `mint_to`, `burn_from` are used
      
  - `technical`:  https://github.com/sora-xor/sora2-network/blob/395-tech-debt-use-assetinfoprovider-instead-of-pallet/pallets/technical/src/lib.rs#L202
     here `transfer_from` and `mint_to` are used     
     
  - `trading-pair`: here only used AssetId type from assets directly
  
  - `vested-rewards`: https://github.com/sora-xor/sora2-network/blob/395-tech-debt-use-assetinfoprovider-instead-of-pallet/pallets/vested-rewards/src/lib.rs#L364
       here `transfer_from` is used
   
   for `xor-fee`, the `burn_from` is used
   
   ## Conclusion:
   -  All before use AssetId type mainly from assets which means removing `assets` dependency from them should be after moving AssetId type to different dependency like `common`
   
   - Also, I think we can add another trait like AssetManager to share the logic of transferring and minting
     functions are : ```burn_from, transfer_from, mint_to, mint, burn, register_from```
     
     
    @wer1st  @vovac12  what do you recommend the next step to be?

Reply:
> All before use AssetId type mainly from assets which means removing assets dependency from them should be after moving AssetId type to different dependency like common

I see 2 optons:
- move `AssetId` into the common config trait
- use `AssetId` as associated type in `AssetInfoProvider` trait

Let's continue the discussion in the dev chat

---
> Also, I think we can add another trait like AssetManager to share the logic of transferring and minting
functions are : burn_from, transfer_from, mint_to, mint, burn, register_from

We have to create another task for that

Reply:
Created a task #983 to remove assets::Config